.github/workflows/main.yml

```diff
@@ -5,68 +5,58 @@ on: [push, pull_request]
 env:
   PYTEST_ADDOPTS: "--color=yes"
 
-jobs:
+permissions:
+  contents: read
 
+jobs:
   test:
-    name: Test - ${{ matrix.python-version }}
+    name: Test - Python ${{ matrix.python-version }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
 
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Install Poetry
-        run: pipx install poetry
-
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
         with:
-          python-version: ${{ matrix.python-version }}
-          cache: "poetry"
-          cache-dependency-path: "pyproject.toml"
+          persist-credentials: false
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
 
       - name: Install dependencies
+        env:
+          UV_PYTHON: "python${{ matrix.python-version }}"
         run: |
-          poetry env use "${{ matrix.python-version }}"
-          poetry run pip install --upgrade libsass
-          poetry install --no-interaction
+          uv sync --frozen --no-sources
 
       - name: Run tests
-        run: poetry run invoke tests
-
+        run: uv run invoke tests
 
   lint:
     name: Lint
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
 
       - name: Validate links in Markdown files
         uses: JustinBeckwith/linkinator-action@v1
         with:
           retry: true
 
-      - name: Install Poetry
-        run: pipx install poetry
-
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.9"
-          cache: "poetry"
-          cache-dependency-path: "pyproject.toml"
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
 
       - name: Install dependencies
+        env:
+          UV_PYTHON: "python3.10"
         run: |
-          poetry env use "3.9"
-          poetry install --no-interaction
+          uv sync --frozen --no-sources
```

Comment:
why not `uv run inv setup` here? Because `inv setup` installs unfrozen dependencies for development, instead of _only_ running
with the committed requirements.

Reply:
Agreed — that is probably a better approach 👍